### PR TITLE
(fix): logback config path seperator and brokerLogDir default value m…

### LINKED ISSUE #6589: logback config ${brokerLogDir} missing default value cause brokerLogDir_IS_UNDEFINED in log path
The issue tracker is used for bug reporting purposes **ONLY** whereas feature request needs to follow the [RIP process](https://github.com/apache/rocketmq/wiki/RocketMQ-Improvement-Proposal). To avoid unnecessary duplication, please check whether there is a previous issue before filing a new one.

It is recommended to start a discussion thread in the [mailing lists](http://rocketmq.apache.org/about/contact/) or [github discussions](https://github.com/apache/rocketmq/discussions) in cases of discussing your deployment plan, API clarification, and other non-bug-reporting issues.
We welcome any friendly suggestions, bug fixes, collaboration, and other improvements.

Please ensure that your bug report is clear and self-contained. Otherwise, it would take additional rounds of communication, thus more time, to understand the problem itself.

Generally, fixing an issue goes through the following steps:
1. Understand the issue reported;
1. Reproduce the unexpected behavior locally;
1. Perform root cause analysis to identify the underlying problem;
1. Create test cases to cover the identified problem;
1. Work out a solution to rectify the behavior and make the newly created test cases pass;
1. Make a pull request and go through peer review;

As a result, it would be very helpful yet challenging if you could provide an isolated project reproducing your reported issue. Anyway, please ensure your issue report is informative enough for the community to pick up. At a minimum, include the following hints:

**BUG REPORT**

1. Please describe the issue you observed:

- What did you do (The steps to reproduce)?
Run broker with default config

- What is expected to see?
Generate log path $user.home/logs/rocketmqlogs/ and all log under the path

- What did you see instead?
Generate log path $user.home/logs/rocketmqlogs/brokerLogDir_IS_UNDEFINED and all broker / proxy load under the path

2. Please tell us about your environment:

3. Other information (e.g. detailed explanation, logs, related issues, suggestions on how to fix, etc):

### PR DESCRIPTION
**Pull request by superhx**
…issing

**Make sure set the target branch to `develop`**

## What is the purpose of the change

fix #6589

## Brief changelog

- replace / with OS unrelated {file.seperator}
- replace ${brokerLogDir} with ${brokerLogDir:-${file.separator}} to set default value when brokerLogDir not set

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml [hunk 1]
```diff
@@ -289,10 +289,10 @@
 
     <appender name="RocketmqCommercialAppender"
               class="ch.qos.logback.core.rolling.RollingFileAppender">
-        <file>${user.home}/logs/rocketmqlogs/${brokerLogDir}/commercial.log</file>
+        <file>${user.home}/logs/rocketmqlogs/${brokerLogDir:-${file.separator}}/commercial.log</file>
```
Comment (aaron-ai) [inline test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml:292]: Why not replace `/` with `${file.separator}` here?

Comment (superhx) [reply test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml:292]: Fixed, please re-review

File: test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml [hunk 2]
```diff
@@ -63,10 +63,10 @@
     <!-- Below is the logger configuration for broker-->
     <appender name="DefaultAppender"
               class="ch.qos.logback.core.rolling.RollingFileAppender">
-        <file>${user.home}/logs/rocketmqlogs/${brokerLogDir}/broker_default.log</file>
+        <file>${user.home}/logs/rocketmqlogs/${brokerLogDir:-${file.separator}}/broker_default.log</file>
         <append>true</append>
         <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
-            <fileNamePattern>${user.home}/logs/rocketmqlogs/otherdays/${brokerLogDir}/broker_default.%i.log.gz</fileNamePattern>
+            <fileNamePattern>${user.home}/logs/rocketmqlogs/otherdays/${brokerLogDir:-${file.separator}}/broker_default.%i.log.gz</fileNamePattern>
             <minIndex>1</minIndex>
```
Comment (aaron-ai) [inline test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml:70]: ditto.

File: test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml [hunk 3]
```diff
@@ -81,10 +81,10 @@
 
     <appender name="RocketmqBrokerAppender_inner"
               class="ch.qos.logback.core.rolling.RollingFileAppender">
-        <file>${user.home}/logs/rocketmqlogs/${brokerLogDir}/broker.log</file>
+        <file>${user.home}/logs/rocketmqlogs/${brokerLogDir:-${file.separator}}/broker.log</file>
         <append>true</append>
         <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
-            <fileNamePattern>${user.home}/logs/rocketmqlogs/otherdays/${brokerLogDir}/broker.%i.log.gz</fileNamePattern>
+            <fileNamePattern>${user.home}/logs/rocketmqlogs/otherdays/${brokerLogDir:-${file.separator}}/broker.%i.log.gz</fileNamePattern>
```
Comment (aaron-ai) [inline test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml:87]: ditto.

**Comment by aaron-ai** [comment]
@superhx Thx for your contribution! please make sure all `/` have been replaced by `${file.separator}`.

File: test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml (hunk 3 above)
Comment (superhx) [reply test/src/test/resources/rmq-proxy-home/conf/logback_proxy.xml]: Done, and confirm there is any /logs exists in logback config xml

**Comment by superhx** [comment]
@aaron-ai Hi，please merge the PR if there is no others need to modify.

File: proxy/src/main/resources/rmq.proxy.logback.xml [hunk 4]
```diff
@@ -229,10 +229,10 @@
 
     <appender name="RocketmqRebalanceLockAppender_inner"
               class="ch.qos.logback.core.rolling.RollingFileAppender">
-        <file>${user.home}/logs/rocketmqlogs/${brokerLogDir}/lock.log</file>
+        <file>${user.home}${file.separator}logs${file.separator}rocketmqlogs${file.separator}${brokerLogDir:-${file.separator}}${file.separator}lock.log</file>
```
Comment (aaron-ai) [inline proxy/src/main/resources/rmq.proxy.logback.xml:232]: If `brokerLogDir` is not set, there will be three consecutive `${file.separator}` here. Have you tested it on multiple operating systems to ensure that the path with three consecutive `${file.separator}` works properly?

Comment (superhx) [reply proxy/src/main/resources/rmq.proxy.logback.xml:232]: there will only be one file.separator in real log path

Comment (aaron-ai) [reply proxy/src/main/resources/rmq.proxy.logback.xml:232]: @superhx Can you provide test results on MacOS/Windows/Linux, or provide evidence about how it works?

Comment (superhx) [reply proxy/src/main/resources/rmq.proxy.logback.xml:232]: Use launch cmd in RocketMQ Quick Start
nohup sh bin/mqnamesrv &
nohup sh bin/mqbroker -n localhost:9876 --enable-proxy &

In my local machine Mac:
image

In Linux:
image


Don't have Windows machine


PS. the proxy logback config path format is the same as rmq.broker.logback.xml

**Review by aaron-ai — APPROVED** [review]
Totally looks solid.

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**